### PR TITLE
Custom costmap converter

### PR DESCRIPTION
**Pull request by PhileasL**
## Purpose

The purpose of this PR is to create a custom costmap_converter that is suitable for our use : more efficient, more accurate. The way we are using costmap_converter package is just to simplify the static costmap as a costmap described by polygons and lines.

## Why a custom plugin ?

In fact the main problem of costmap_converter package in our case is that in edges of the map, the package assumes that it can simplifies the 2 intercepted lines by a triangle. WRONG ! In that way, it creates a virtual lethal obstacle on the game area, we don't want this kind of behavior. Plus, it uses a heavy algorithm that computes and tries to decompose the static costmap into polygons several times per second. 
Considering the static costmap as a layer that will never change, we want to reduce the computational time by decomposes the map only one time at launching, and avoid the apparition of virtual lethal obstacles on the game area.

## How does it works ?

In the initialization sequence of costmap_converter_custom, the plugin gets the data of "static_map_lines" in the configuration yaml file. This parameter is a string of a vector of vectors of floats that is a description of costmap lines described as following : 
`static_map_lines: "[ [pt1_start_x, pt1_start_y, pt1_end_x, pt1_end_y],[pt2_start_x, pt2_start_y, pt2_end_x, pt2_end_y], ... ]"
`
Then it performs a naive algorithm that creates the lines from theses coordinates and store them into a global variable that will never change as a PolygonContainerPtr type.
When needed (parameter `costmap_converter_rate` in yaml), the costmap_converter package get this PolygonContainerPtr, converts it as obstacles and publish them in order to communicate the teb_local_planner the lethal obstacles.

## Problem

By default in teb_local_planner, the obstacles are 10 cm thick and in reality, they are just 2 cm thick. And that is a constant value in the teb_local_planner obstacle, plus the lifetime of the obstacle isn't long enough (2 seconds). 
In order to make the plugin working as expected, I modify teb_local_planner to set the obstacles as 2 cm thick. [See those changes here.](https://github.com/robotique-ecam/teb_local_planner/commit/bd684288eb64aa86a144d2ff84e8c41b700e50bf). So, in order to make it fully functional, you have to update your teb_local_planner.

#### Observations

The plugin works fine, but sometimes teb_local_planner can still find a way through a wall, I don't really know why for the moment, but I've never seen the robot following a plan that is going through a wall.

#### Extras

- Map package has been a bit simplified, no need to put the path in argument anymore in map.py. 
- I've found parameters for teb_local_planner to get rid of oscillations.

### CODE REVIEW

File: src/modules/map/map/map.py [hunk 1]
```diff
@@ -53,23 +54,23 @@ def creamap():
...
+            (cm_pix(210, reso), cm_pix(186 + reso, reso)),
         ),
         noir,
         1,
     )
     draw.line(
         (
             (cm_pix(150, reso), cm_pix(largeur_cm, reso)),
-            (cm_pix(150, reso), cm_pix(170, reso)),
+            (cm_pix(150, reso), cm_pix(170 + reso, reso)),
```
Comment (PhileasL) [inline src/modules/map/map/map.py:73]: I think the offset is still one cm, to confirm before rebase.

File: src/modules/robot/param/robot.in.yml [hunk 2]
```diff
@@ -114,6 +116,15 @@ controller_server:
       # Miscellaneous
       map_frame: "map"
 
+costmap_converter:
+  ros__parameters:
+    use_sim_time: false
+    # Decomposition of the map into 10 cm thickness lines
```
Comment (PhileasL) [inline src/modules/robot/param/robot.in.yml:122]: We're not at 10 cm anymore but 2.

**Review by PhileasL — COMMENTED** [review]
Change the name of the class, it isn't CostmapToPolygonsDBSMCCH at all.

File: src/modules/robot/param/robot.in.yml (hunk 2 above)
Comment (PhileasL) [reply src/modules/robot/param/robot.in.yml]: [CI foxy-devel branch + 2cm thickness](https://github.com/robotique-ecam/cdfr/pull/35/commits/3c03077757d08c4f468d0af7c1191833a79673d9)

**Comment by PhileasL** [comment]
Updated CI branch for teb_local_planner **foxy-devel**

File: .github/workflows/ci_ws.repos [hunk 3]
```diff
@@ -6,4 +6,4 @@ repositories:
   teb_local_planner:
     type: git
     url: https://github.com/robotique-ecam/teb_local_planner.git
-    version: foxy
```
Comment (3wnbr1) [inline .github/workflows/ci_ws.repos:9]: key must be change to invalidate ros_ws cache and rebuild

https://github.com/robotique-ecam/cdfr/blob/7838232d9422ad673d1401377f13d021f389c8c2/.github/workflows/main.yml#L37

**Review by 3wnbr1 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci_ws.repos (hunk 3 above)
Comment (3wnbr1) [reply .github/workflows/ci_ws.repos:9]: Otherwise, CI will use its cache and ignore your entry

Comment (PhileasL) [reply .github/workflows/ci_ws.repos:9]: Done, [CI cache key update ](https://github.com/robotique-ecam/cdfr/pull/35/commits/326a64e042b68773e58bb4563afe44d9aeabc88b)

**Review by 3wnbr1 — APPROVED** [review]

**Comment by 3wnbr1** [comment]
Nice work there, just waiting for the CI Status 👍

**Pull request merged**